internal/controller/management/project-configs/project_configs.go

```diff
@@ -0,0 +1,257 @@
+		conditions.Set(status, &metav1.Condition{
+			Type:               kargoapi.ConditionTypeReady,
+			Status:             metav1.ConditionFalse,
+			Reason:             "Syncing",
+			Message:            "Failed to ensure project config webhook receivers: " + err.Error(),
+			ObservedGeneration: projectConfig.GetGeneration(),
+		})
+		return *status, true, fmt.Errorf("error ensuring webhook receivers: %w", err)
+	}
+	status.WebhookReceivers = projectConfig.Status.WebhookReceivers
+
+	conditions.Delete(status, kargoapi.ConditionTypeReconciling)
+	conditions.Set(status, &metav1.Condition{
+		Type:               kargoapi.ConditionTypeReady,
+		Status:             metav1.ConditionTrue,
+		Reason:             "Synced",
+		Message:            "ProjectConfig is synced and ready for use",
+		ObservedGeneration: projectConfig.GetGeneration(),
+	})
+	return *status, false, nil
+}
+
+func (r *reconciler) ensureWebhookReceivers(
+	ctx context.Context,
+	pc *kargoapi.ProjectConfig,
+) error {
+	logger := logging.LoggerFromContext(ctx)
+
+	if pc.Spec.WebhookReceiverConfigs == nil {
+		logger.Debug("ProjectConfig does not have any receiver configurations")
+		return nil
+	}
+
+	logger.Debug("ensuring receivers",
+		"receiver-configs", len(pc.Spec.WebhookReceiverConfigs),
+	)
+
+	for _, rc := range pc.Spec.WebhookReceiverConfigs {
+		if rc.GitHub != nil {
+			if err := r.ensureGitHubWebhookReceiver(ctx, pc, rc); err != nil {
+				logger.Error(err, "error ensuring GitHub webhook receiver",
+					"receiver-config", rc,
+				)
+				return fmt.Errorf("error ensuring GitHub webhook receiver: %w", err)
+			}
+		}
+	}
+	return nil
+}
+
+func (r *reconciler) ensureGitHubWebhookReceiver(
+	ctx context.Context,
+	pc *kargoapi.ProjectConfig,
+	rc kargoapi.WebhookReceiverConfig,
+) error {
+	logger := logging.LoggerFromContext(ctx)
+	var secret corev1.Secret
+	err := r.client.Get(
+		ctx,
+		types.NamespacedName{
+			Namespace: pc.Namespace,
+			Name:      rc.GitHub.SecretRef.Name,
+		},
+		&secret,
+	)
+	if err != nil {
+		logger.Error(err, "github secret not found",
+			"github-secret", rc.GitHub.SecretRef.Name,
+		)
+		if kubeerr.IsNotFound(err) {
+			return fmt.Errorf(
+				"secret-reference %q in namespace %q not found",
+				rc.GitHub.SecretRef.Name, pc.Namespace,
+			)
+		}
+		return fmt.Errorf(
+			"error getting webhook receiver secret-reference %q in project namespace %q: %w",
+			rc.GitHub.SecretRef.Name, pc.Name, err,
+		)
+	}
+	logger.Debug("secret found", "secret", secret.Name)
+
+	token, ok := secret.StringData["token"]
+	if !ok {
+		logger.Error(err, "'token' key not found in secret data")
+		return fmt.Errorf(
+			"key 'token' not found in secret %q for project config %q",
+			rc.GitHub.SecretRef.Name, pc.Name,
+		)
+	}
+	pc.Status.WebhookReceivers = append(pc.Status.WebhookReceivers,
+		kargoapi.WebhookReceiver{
+			Path: external.GenerateWebhookPath(
+				pc.Name,
+				kargoapi.WebhookReceiverTypeGitHub,
+				token,
+			),
+		},
+	)
```

Comment:
I do not think this handles the deletion of a webhook configuration properly? I think for this to work, you would need to start with an empty slice every reconciliation.